src/main/java/commission/service/SaleServiceImpl.java

```diff
@@ -95,14 +95,23 @@ public boolean updateSalePercentage(long id, Sale sale) {
         boolean isUpdated = false;
         var temp = findSaleById(id);
         if (temp.isEmpty()) {
-            System.out.println("ID does not exists");
+            logger.error("ID not found");
         } else {
-            saleDao.updateSalePercentage(id, sale);
+            var currentPrice = temp.get().price();
+            saleDao.updateSalePercentage(id, sale, commissionPerSale(currentPrice, sale.percentage()) );
             isUpdated = true;
         }
         return isUpdated;
     }
 
 
+    public double commissionPerSale(Sale sale){
+        return sale.price() * sale.percentage() / 100 ;
+    }
+
+    public double commissionPerSale(double valueA, double valueB){
```

Comment:
This is a small method and easy to understand it within it's context, but it's still good practice to rather names these variables more appropriately. It will help someone else reading the code to understand and if it should be modified at some point make it easier to reason about.

Reply:
Now that you point it out, yes I dont know what I was thinking when I was naming the parameters.